File: 프로그래머스/1/12928. 약수의 합/약수의 합.py
import math
def solution(n):
    s = math.floor(math.sqrt(n)) #제곱근 구하기
    print(s)
    answer = 0
    if n==0:
        return 0
    elif n==1:
        return 1
    for i in range(1,s+1):
        if n%i==0:
            if i == n//i:
                answer +=i
            else:
                answer += i + n//i
    return answer

File: 프로그래머스/1/12928. 약수의 합/약수의 합.java
import java.util.*; //arraylist, set, HashSet을 사용하기 위해 util함수가 필요

class Solution {
    public int solution(int n) {
        ArrayList<Integer> l = new ArrayList<>();
        for(int i = 1; i<= (int) Math.sqrt(n); i++){
               if (n % i == 0){
                   l.add(i);
                   l.add(n/i);
               }
        }
        Set<Integer> set = new HashSet<>(l);
        int result= 0;
        for(int s : set){
            result += s;
        }
        return result;
    }
}